### DISCUSSION TITLE
Using @Environment will crash in Preview

### DISCUSSION THREAD

**Discussion by zhuscat** [pointfreeco/swift-perception#115] [General]
### Description

Here is my code:
```swift
import SwiftUI
import Perception

@Perceptible
class HelloModel {
  var msg: String = "Hello, World"
}

struct HelloView: View {
  @Environment(HelloModel.self) var model

    var body: some View {
      WithPerceptionTracking {
        Text(model.msg)
      }
    }
}

#Preview {
    HelloView()
      .environment(HelloModel())
}

```

It crashes in preview:

```
== PAUSED REASON:

    processCrashed



== PREVIEW UPDATE ERROR:

    CrashReportError: Fatal Error in Environment.swift
    
    fito crashed due to fatalError in Environment.swift at line 54.
    
    No perceptible object of type HelloModel found. A View.environment(_:) for HelloModel may be missing as an ancestor of this view.
```

### Checklist

- [X] I have determined that this bug is not reproducible using Swift's observation tools. If the bug is reproducible using the `@Observable` macro or another tool from the `Observation` framework, please [file it directly with Apple](https://github.com/apple/swift/issues/new/choose).
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-perception/issues) or [discussion](https://github.com/pointfreeco/swift-perception/discussions).

### Expected behavior

No crash happened

### Actual behavior

Crash happened

### Steps to reproduce

Described in description

### Perception version information

1.4.1

### Destination operating system

iOS 16.0

### Xcode version information

Version 16.1 (16B40)

**Comment by zhuscat** [comment]
Using `PreviewProvider` is working:

```
struct HelloView_Previews: PreviewProvider {
    static var previews: some View {
      HelloView()
        .environment(HelloModel())
    }
}
```

**Comment by mbrandonw** [comment]
Hi @zhuscat, since this works with preview provides I'm inclined to consider this an Apple bug. If someone finds a work around we will happily accept a PR, but this isn't something we personally have time to investigate.

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!
